playbooks/roles/certificates/tasks/client.yml

```diff
@@ -0,0 +1,39 @@
+---
+- name: Generate RSA keys for the clients
```

Comment:
I think this role could use a little documentation, since it's designed to be reused. When I got to this line, I said "which clients?" too. By now I've figured out what you're doing, but it wasn't immediately obvious to me at the time.

Reply:
@alimakki How about writing a tiny file in `docs/` on certificate generation to go with this? Or maybe a mention in `CONTRIBUTING.md` "How to generate certificates"?

Reply:
Explicitly showing it as a tree might help. TBH, I am confused by all of the certs wandering around.

```
Root CA: (dead)
   \__ gateway HTTPS cert (unless LE; see below)
   \__ Foobar server cert
   \__ Foobar CA
        \__ foobar client 1

Let's Encrypt Authority:
   \__ poem-walk.example.com HTTPS cert
```

Reply:
I think presently there are multiple Root CAs being generated:

```
Web Root CA: 
   \__ gateway HTTPS cert (unless LE; see below)

Foobar Root CA: 
   \__ Foobar server cert
   \__ foobar client 1

ISRG X1:
   \__ Let's Encrypt Authority X3
        \__ poem-walk.example.com HTTPS cert
```

We can improve on the structure of the internal PKI as a follow-up. The way you're imagining it is closer to how it probably should be (though I would make one root CA and then issue constrained intermediates per-service).

Reply:
@cpu, I'll try to write up something in a separate .md file, should be easier to hunt down if someone were to look for it.